src/util/delegate.js

```diff
@@ -0,0 +1,28 @@
+/**
+ * Generate a bunch of accessors to proxy properties through a certain subobject
+ * @param {Object} options
+ * @param {Object} options.from - The object to define the delegated properties on
+ * @param {string} options.to - The name of the subobject property to proxy through
+ * @param {string[]} options.properties - Array of property names to delegate
+ * @param {Object<string, PropertyDescriptor>} options.descriptors - Property descriptors for each property
+ */
+export function delegate ({from, to, properties, descriptors}) {
+	for (let prop of properties) {
+		let sourceDescriptor = descriptors[prop];
+		let descriptor = {
+			get () {
+				return this[to][prop];
+			},
+			...sourceDescriptor,
+			configurable: true,
+		};
+
+		if (sourceDescriptor.writable || sourceDescriptor.set) {
+			descriptor.set = function (value) {
```

Comment:
```suggestion
			delete descriptor.value;
			delete descriptor.writable;

			descriptor.set = function (value) {
```

Otherwise, some descriptors might throw on line 26. For example, `checkValidity` is writable and has a value. After adding accessors, we make the descriptor invalid. 

<img width="199" height="150" alt="Image" src="https://github.com/user-attachments/assets/43c0b006-e4f7-4bd4-a1d2-d173da1a9416" />